src/renderer/lib/utils.ts

```diff
@@ -61,3 +62,12 @@ export const humanReadableBytes = (bytes: number): string => {
 
   return `0 ${sizes[0]}`;
 };
+
+export const isSubdir = (parentPath: string, dirPath: string) => {
+  const relativePath = path.relative(parentPath, dirPath);
+  return (
+    relativePath && // value is "" when paths are identical
+    !relativePath.startsWith("..") &&
+    !path.isAbsolute(relativePath) // covers case where different drives are used e.g. 'C:\\A' vs 'D:\\A\\B'
+  );
+};
```

Comment:
This logic isn't perfect, as there's still the [edge case where a user creates a directory that starts with `".."`](https://stackoverflow.com/a/45242825). However because this seems unlikely, I think it would be okay to settle with this for now. Open to hearing other people's thoughts on this though!

Reply:
I think that's fine, not the worst situation to deal with.